lib/src/solver/report.dart

```diff
@@ -130,6 +133,26 @@ class SolveReport {
     }
   }
 
+  /// Displays a warning about any discontinued packages depended upon.
+  Future<void> _reportDiscontinued() async {
+    for (final id in _result.packages) {
```

Comment:
Also maybe we should do:

```dart
await Future.wait(_result.packages.map((id) async {
  ...
}));
```

So that we get more concurrency in cases where this information isn't in the cache.